lib/net.js

```diff
@@ -669,14 +669,12 @@ Socket.prototype._destroy = function(exception, cb) {
 };
 
 Socket.prototype._getpeername = function() {
-  if (!this._peername) {
-    if (!this._handle || !this._handle.getpeername) {
-      return {};
-    }
-    const out = {};
-    const err = this._handle.getpeername(out);
-    if (err) return {};  // FIXME(bnoordhuis) Throw?
-    this._peername = out;
+  if (!this._handle || !this._handle.getpeername) {
+    return this._peername || {};
+  } else if (!this._peername) {
+    this._peername = {};
```

Comment:
This is a functional change in that before errors can be transient, i.e., the first call can return an empty object  while the second call can return a full-fledged address object. Now errors are no longer transient because the empty object is cached.

It's a probably inconsequential change in behavior but I'd label it semver-major anyway.